src/extensions/StakerPermitAndStakeUpgradeable.sol

```diff
@@ -11,15 +11,24 @@ import {IERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/IERC2
 /// The permit functionality is used in conjunction with staking operations, improving UX by
 /// enabling users to approve and stake tokens in a single transaction. Note that this extension
 /// requires the stake token to support EIP-2612 permit functionality.
-abstract contract StakerPermitAndStake is Staker {
+abstract contract StakerPermitAndStakeUpgradeable is StakerUpgradeable {
   /// @notice Thrown if an inheritor misconfigures the staking token on deployment.
-  error StakerPermitAndStake__UnauthorizedToken();
+  error StakerPermitAndStakeUpgradeable__UnauthorizedToken();
 
   /// @param _permitToken The token that is used for staking, which must support EIP-2612. It also
```

Comment:
Missing `/// must be the same as the parent Staker's STAKE_TOKEN.`
- Add natspec